internal/local_preview.go

```diff
@@ -45,8 +46,10 @@ func MoveToRepositoryRoot() error {
 	}
 }
 
-// GetIgnoreMatcherFn creates an ignore-matcher for archiving purposes. Respects gitignore and terraformignore.
-func GetIgnoreMatcherFn(ctx context.Context) (func(filePath string) bool, error) {
+// GetIgnoreMatcherFn creates an ignore-matcher for archiving purposes
+// This function respects gitignore and terraformignore, and
+// optionally if a projectRoot is provided it only include files from this root
+func GetIgnoreMatcherFn(ctx context.Context, projectRoot string) (func(filePath string) bool, error) {
```

Comment:
There is the open question I have after looking at this code - do we want to be doing recursive lookups of the .gitignore files or just look at the files in the root of the repo? Seems like this would ignore the local directory and go straight to the root.